File: src/main/java/org/lanternpowered/server/network/entity/EntityProtocolTypes.java
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.entity;

import org.lanternpowered.server.entity.LanternEntity;
import org.lanternpowered.server.entity.player.LanternPlayer;
import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;

public final class EntityProtocolTypes {

    public static final EntityProtocolType<LanternEntity> ARMOR_STAND = dummy("ARMOR_STAND");

    public static final EntityProtocolType<LanternEntity> BAT = dummy("BAT");

    public static final EntityProtocolType<LanternEntity> CHICKEN = dummy("CHICKEN");

    public static final EntityProtocolType<LanternEntity> ENDER_DRAGON = dummy("ENDER_DRAGON");

    public static final EntityProtocolType<LanternEntity> ENDERMITE = dummy("ENDERMITE");

    public static final EntityProtocolType<LanternEntity> EXPERIENCE_ORB = dummy("EXPERIENCE_ORB");

    public static final EntityProtocolType<LanternEntity> GIANT = dummy("EXPERIENCE_ORB");

    public static final EntityProtocolType<LanternEntity> HUMAN = dummy("HUMAN");

    public static final EntityProtocolType<LanternEntity> HUSK = dummy("HUSK");

    public static final EntityProtocolType<LanternEntity> IRON_GOLEM = dummy("IRON_GOLEM");

    public static final EntityProtocolType<LanternEntity> ITEM = dummy("ITEM");

    public static final EntityProtocolType<LanternEntity> LIGHTNING = dummy("LIGHTNING");

    public static final EntityProtocolType<LanternEntity> MAGMA_CUBE = dummy("MAGMA_CUBE");

    public static final EntityProtocolType<LanternEntity> PAINTING = dummy("PAINTING");

    public static final EntityProtocolType<LanternEntity> PIG = dummy("PIG");

    public static final EntityProtocolType<LanternPlayer> PLAYER = dummy("PLAYER");

    public static final EntityProtocolType<LanternEntity> RABBIT = dummy("RABBIT");

    public static final EntityProtocolType<LanternEntity> SHEEP = dummy("SHEEP");

    public static final EntityProtocolType<LanternEntity> SILVERFISH = dummy("SILVERFISH");

    public static final EntityProtocolType<LanternEntity> SLIME = dummy("SLIME");

    public static final EntityProtocolType<LanternEntity> SNOWMAN = dummy("SNOWMAN");

    public static final EntityProtocolType<LanternEntity> VILLAGER = dummy("VILLAGER");

    public static final EntityProtocolType<LanternEntity> ZOMBIE = dummy("ZOMBIE");

    public static final EntityProtocolType<LanternEntity> ZOMBIE_VILLAGER = dummy("ZOMBIE_VILLAGER");

    public static final EntityProtocolType<LanternEntity> HORSE = dummy("HORSE");

    public static final EntityProtocolType<LanternEntity> DONKEY = dummy("DONKEY");

    public static final EntityProtocolType<LanternEntity> LLAMA = dummy("LLAMA");

    private static <E extends LanternEntity> EntityProtocolType<E> dummy(String name) {
        //noinspection unchecked
        return DummyObjectProvider.createFor(EntityProtocolType.class, name);
    }

    private EntityProtocolTypes() {
    }
}

File: src/main/kotlin/org/lanternpowered/server/network/entity/EntityProtocolTypes.kt
/*
 * Lantern
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) SpongePowered <https://www.spongepowered.org>
 * Copyright (c) contributors
 *
 * This work is licensed under the terms of the MIT License (MIT). For
 * a copy, see 'LICENSE.txt' or <https://opensource.org/licenses/MIT>.
 */
package org.lanternpowered.server.network.entity

import org.lanternpowered.api.key.minecraftKey
import org.lanternpowered.api.registry.CatalogRegistry
import org.lanternpowered.api.registry.require
import org.lanternpowered.server.entity.LanternEntity
import org.lanternpowered.server.entity.player.LanternPlayer

object EntityProtocolTypes {
    val ARMOR_STAND: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("armor_stand"))
    val BAT: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("bat"))
    val CHICKEN: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("chicken"))
    val ENDER_DRAGON: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("ender_dragon"))
    val ENDERMITE: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("endermite"))
    val EXPERIENCE_ORB: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("experience_orb"))
    val GIANT: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("giant"))
    val HUMAN: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("human"))
    val HUSK: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("husk"))
    val IRON_GOLEM: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("iron_golem"))
    val ITEM: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("item"))
    val LIGHTNING: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("lightning"))
    val MAGMA_CUBE: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("magma_cube"))
    val PAINTING: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("painting"))
    val PIG: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("pig"))
    val PLAYER: EntityProtocolType<LanternPlayer> = CatalogRegistry.require(minecraftKey("player"))
    val RABBIT: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("rabbit"))
    val SHEEP: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("sheep"))
    val SILVERFISH: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("silverfish"))
    val SLIME: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("slime"))
    val SNOWMAN: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("snowman"))
    val VILLAGER: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("villager"))
    val ZOMBIE: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("zombie"))
    val ZOMBIE_VILLAGER: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("zombie_villager"))
    val HORSE: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("horse"))
    val DONKEY: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("donkey"))
    val LLAMA: EntityProtocolType<LanternEntity> = CatalogRegistry.require(minecraftKey("llama"))

}